Data/Scripts/001_Technical/002_RubyUtilities.rb

```diff
@@ -162,6 +159,145 @@ def self.move(source, destination)
   end
 end
 
+#===============================================================================
+#  Extensions for the `Color` class
+#===============================================================================
+class Color
+  # alias for old constructor
+  alias init_original initialize unless self.private_method_defined?(:init_original)
+  #-----------------------------------------------------------------------------
+  # new constructor accepts RGB values as well as a hex number or string value
+  #-----------------------------------------------------------------------------
+  def initialize(*args)
+  	pbPrintException("Wrong number of arguments! At least 1 is needed!") if args.length < 1
+  	if args.length == 1
+      if args[0].is_a?(Fixnum)
+        hex = args[0].dup.to_s(16)
+      elsif args[0].is_a?(String)
+        hex = args[0].dup
+        hex.gsub!("#", "") if hex.include?("#")
```

Comment:
You could call `gsub!` even if `hex` didn't include "#". I doubt `gsub!` is significantly slower at finding a character than `include?` is.

Reply:
I think I may have pushed some older version by mistake, this part should just be `hex.delete('#')` to work properly.